include/rovio/RovioInterface.hpp

```diff
@@ -78,13 +78,23 @@ class RovioInterface {
   virtual bool processImageUpdate(const int camID, const cv::Mat &cv_img,
                                   const double time_s) = 0;
 
+  // 6DoF pose-based localization.
   virtual bool processGroundTruthUpdate(const Eigen::Vector3d &JrJV, const QPD &qJV,
                                         const double time_s) = 0;
 
   virtual bool processGroundTruthOdometryUpdate(
       const Eigen::Vector3d &JrJV, const QPD &qJV,
       const Eigen::Matrix<double, 6, 6> &measuredCov, const double time_s) = 0;
 
+  // Localization landmark-based localization.
+  virtual bool processLocalizationLandmarkUpdates(
+      const int camID, const Eigen::Matrix2Xd& keypoint_observations,
+      const Eigen::Matrix3Xd& I_landmarks, const double time_s) = 0;
```

Comment:
just came to my mind - I could still see some issues with accuracy with the last digits of doubles, even when using the new 'start from zero' logic for rovio timestamps, how is the timestamp used? does it matter if it's synced well with images?

Reply:
discussed offline